### ISSUE TITLE
Zenject support for UWP? 

### ISSUE THREAD

**Issue by nehmebilal** [ssannandeji/Zenject-2019#110]
Hi,

I am wondering if Zenject supports UWP. For instance, I know that `Reflection.Emit` is not allowed in UWP, which makes many libraries incompatible. 

I'd like to use Zenject in Unity but I'd also like to build my app as a UWP app.

Thanks

**Comment by svermeulen** [comment]
UWP should be supported by Zenject.  I just tested this recently using both Universal 8.1 and Universal 10.  There was some special case handling necessary to support this but should work now

Also, Zenject doesn't use Reflection.Emit so that's not a problem.

**Comment by nehmebilal** [comment]
Ok great thanks. I noticed that you have listed "Auto-Mocking using the Moq library" as a feature which uses `Reflection.Emit`. How does that play with UWP? Is it an optional component?

**Comment by svermeulen** [comment]
Yeah exactly, that's optional.  By default that's not included

**Comment by nehmebilal** [comment]
Thanks @svermeulen. I'll close this for now and reopen if we hit any issue.

**Comment by nehmebilal** [comment]
I just tried Zenject in UWP and I get compile errors for all code that use Zenject.

For instance, for the `[Inject]` attribute, I guess the following error:

```
error CS0012: The type 'Attribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=4.0.20.0, Culture=neutral, PublicKeyToken=null'
```

The Zenject references are there in the UWP project but something is not right.

**Comment by svermeulen** [comment]
Is this for "Universal 10" SDK? Also what version of unity are you using?

**Comment by svermeulen** [comment]
I just tried building the VS UWP Universal 10 project using the Asteroids sample project and it built successfully.  (Unity 5.3.4f1)

**Comment by nehmebilal** [comment]
Yes, Universal 10. I'm using Unity 5.3.1p5. You built the `Universal Windows` project in visual studio right?

**Comment by svermeulen** [comment]
Yes.  Using the sample Asteroids project in Zenject, I can change to the Universal 10 platform, check "Unity C# Projects", run a build, open it in VS 2015, and build it successfully.

**Comment by nehmebilal** [comment]
Ok thanks. I'll try with the newest Unity and let you know.

**Comment by nehmebilal** [comment]
Ok so It works when I install Zenject as code but I couldn't get it to work using the Zenject binaries.

**Comment by nehmebilal** [comment]
Seems like 4.0 is not building with UWP. `Type.HasAttribute` was not found (newly introduced to `DiContainer`).

**Comment by svermeulen** [comment]
Thanks - I'll take a look before replacing Zenject 3.  The new convention binding stuff will surely break this as well

**Comment by svermeulen** [comment]
Should be fixed now in version 4.1, if not let me know
